pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -278,3 +299,39 @@ func getPodIndexerByOwnerRefs(indexName string, handle frameworktypes.Handle) (c
 
 	return indexer, nil
 }
+
+func getNamespacesListByLabelSelector(indexName string, labelSelector *metav1.LabelSelector, handle frameworktypes.Handle) (cache.Indexer, error) {
+	nsInformer := handle.SharedInformerFactory().Core().V1().Namespaces().Informer()
+	indexer := nsInformer.GetIndexer()
+
+	// do not reinitialize the indexer, if it's been defined already
+	for name := range indexer.GetIndexers() {
+		if name == indexName {
+			return indexer, nil
+		}
+	}
+
+	if err := nsInformer.AddIndexers(cache.Indexers{
+		indexName: func(obj interface{}) ([]string, error) {
```

Comment:
I wonder whether it would be computationally less expensive to get a namespace object from the cache and check whether `defaultEvictorArgs.NamespaceLabelSelector`  matches the namespace. Compared to adding an index.

Reply:
I've simplified the code (use Lister instead of Indexer).